common/persistence/sql/sqlplugin/postgresql/visibility.go

```diff
@@ -45,10 +45,12 @@ var (
 	templateUpsertWorkflowExecution = fmt.Sprintf(
 		`INSERT INTO executions_visibility (%s)
 		VALUES (%s)
-		%s`,
+		%s
+		WHERE executions_visibility.%s < EXCLUDED.%s`, // This line is to ensure that no update occurs if the version is behind the saved version.
```

Comment:
The `WHERE` part is part of the `ON CONFLICT` expression. Please put this together in the `buildOnDuplicateKeyUpdate` function.

Reply:
Ok, I tossed it here only because it references the table, but if you feel like the function should represent the full expression, I'm fine with that.

Reply:
Do you need to reference the table?

Reply:
Yes. We got an error back from postgress saying  `column reference \"_version\" is ambiguous`